Context propagation documentation (#9554)
micronaut-projects/micronaut-core: src/main/docs/guide/languageSupport/kotlin/kotlinContextPropagation.adoc

For more detailed information on how to use the library you can find at the official link:https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-reactor/kotlinx.coroutines.reactor/-reactor-context/index.html[documentation].

> NOTE: Since Micronaut framework 4 it's recommended to use the latest <<contextPropagation, Context Propagation API>>. The api:core.propagation.ThreadPropagatedContextElement[] is inspired by Kotlin Coroutines propagation API element `kotlinx.coroutines.ThreadContextElement` and acts in the same way by restoring thread locals.

Reviewer: ```suggestion
NOTE: Since Micronaut framework 4, we recommend using the latest <<contextPropagation, Context Propagation API>>. The api:core.propagation.ThreadPropagatedContextElement[] is inspired by Kotlin Coroutines propagation API element `kotlinx.coroutines.ThreadContextElement` and acts similarly by restoring thread locals.
```
